pkg/sql/privilege/privilege_test.go

```diff
@@ -86,3 +87,33 @@ func TestByDisplayNameHasAllPrivileges(t *testing.T) {
 		require.Equal(t, kind, resolvedKind)
 	}
 }
+
+// TestModifyPrivHasCorrespondingViewPriv checks that each MODIFY privilege has
+// a corresponding VIEW privilege.
+func TestModifyPrivHasCorrespondingViewPriv(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	// If you are adding a new MODIFY privilege that does not need a corresponding
+	// VIEW privilege, you must add it to the exceptions below. Verify the
+	// exception with the SQL Foundations or Product Security teams.
+	modifyExceptions := map[privilege.Kind]struct{}{
+		privilege.MODIFYSQLCLUSTERSETTING: {}, // covered by VIEWCLUSTERSETTING
+	}
+
+	for _, modifyPriv := range privilege.AllPrivileges {
```

Comment:
(super?)nit: `modifyPriv` might be a little misleading before we have completed the `HasPrefix` check below